src/DynamoCore/Core/Threading/UpdateGraphAsyncTask.cs

```diff
@@ -102,6 +103,32 @@ protected override void HandleTaskCompletionCore()
             }
         }
 
+        protected override AsyncTask.TaskMergeInstruction CanMergeWithCore(AsyncTask otherTask)
+        {
+            var theOtherTask = otherTask as UpdateGraphAsyncTask;
+            if (theOtherTask == null)
+                return base.CanMergeWithCore(otherTask);
+
+
+            // Comparing to another UpdateGraphAsyncTask, verify that they are updating
+            // a similar set of nodes
+
+            // Other node is either equal or a superset of this task
+            if (ModifiedNodes.All(x => theOtherTask.ModifiedNodes.Contains(x)))
+            {
+                return TaskMergeInstruction.KeepOther;
+            }
+
+            // This node is a superset of the other
+            if (theOtherTask.ModifiedNodes.All(x => ModifiedNodes.Contains(x)))
+            {
+                return TaskMergeInstruction.KeepThis;
+            }
+
+            // They're different, keep both
+            return TaskMergeInstruction.KeepBoth;
+        }
```

Comment:
Thanks for the update, @lukechurch, this looks great! :)

Another minor thing is, as @ke-yu pointed out, `HashSet.IsSupersetOf` can probably do this *better* but for graph nodes (which are not in huge numbers) this should be fine. LGTM!

Sorry for being late on the review anyway.